hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/ITestS3APrefetchingLruEviction.java

```diff
@@ -172,6 +174,8 @@ public void testSeeksWithLruEviction() throws Throwable {
     LambdaTestUtils.eventually(TIMEOUT_MILLIS, INTERVAL_MILLIS, () -> {
       LOG.info("IO stats: {}", ioStats);
       verifyStatisticGaugeValue(ioStats, STREAM_READ_BLOCKS_IN_FILE_CACHE, 0);
+      assertThatStatisticCounter(ioStats,
+          STREAM_EVICT_BLOCKS_FROM_FILE_CACHE).isGreaterThanOrEqualTo(5);
```

Comment:
since this is `eventually()`, should we set the number of evictions that would be happening eventually rather than a least number?

Reply:
there is bit of an async behavior that will likely make the test flaky even with eventually because the order in which lazy seek is going to trigger block cache is not guaranteed and that can still sometimes hurt the exact number that we are trying to match.
i really wish we could do this but it's still lazy seek and verifying the exact order of seeks leading to cache load is not guaranteed every time (some of the problems are also being discussed on unbuffer PR).